agent/taskresource/firelens/firelensconfig_windows.go

```diff
@@ -0,0 +1,82 @@
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package firelens
+
+import (
+	generator "github.com/awslabs/go-config-generator-for-fluentd-and-fluentbit"
+)
+
+const (
+	// S3ConfigPathFluentd and S3ConfigPathFluentbit are the paths where we bind mount the config downloaded from S3 to.
+	S3ConfigPathFluentd   = `C:\data\fluentd\etc\`
+	S3ConfigPathFluentbit = `C:\data\fluent-bit\etc\`
+
+	// inputBindValue is the value for specifying host on Windows this always binds to 0.0.0.0
+	inputBridgeBindValue = "0.0.0.0"
+
+	// inputAWSVPCBindValue is the value for specifying host for AWSVPC mode.
+	inputAWSVPCBindValue = "0.0.0.0"
+)
+
+// Specify log stream input, which is a unix socket that will be used for communication between the Firelens
+// container and other containers.
+// not supported on Windows
+func (firelens *FirelensResource) addSocketInput(config generator.FluentConfig) {
+
+}
+
+// Specify log stream input of tcp socket kind that can be used for communication between the Firelens
+// container and other containers if the network is bridge or awsvpc mode. Also add health check sections to support
+// doing container health check on firlens container for these two modes.
+func (firelens *FirelensResource) addTCPInput(config generator.FluentConfig) {
+	if firelens.networkMode == bridgeNetworkMode || firelens.networkMode == awsvpcNetworkMode || firelens.networkMode == "" {
+		var inputMap map[string]string
+		var inputName, inputBindValue string
+		if firelens.networkMode == bridgeNetworkMode || firelens.networkMode == "" {
+			inputBindValue = inputBridgeBindValue
+		} else if firelens.networkMode == awsvpcNetworkMode {
+			inputBindValue = inputAWSVPCBindValue
+		}
+		if firelens.firelensConfigType == FirelensConfigTypeFluentd {
+			inputMap = map[string]string{
+				inputPortOptionFluentd: inputPortValue,
+				inputBindOptionFluentd: inputBindValue,
+			}
+			inputName = inputNameForward
+		} else {
+			inputName = inputNameForward
+			inputMap = map[string]string{
+				inputPortOptionFluentbit:   inputPortValue,
+				inputListenOptionFluentbit: inputBindValue,
+			}
+		}
+		config.AddInput(inputName, "", inputMap)
+		firelens.addHealthcheckSections(config)
+	}
+}
+
+// addHealthcheckSections adds a health check input section and a health check output section to the config.
+func (firelens *FirelensResource) addHealthcheckSections(config generator.FluentConfig) {
+
+}
+
+func (firelens *FirelensResource) getS3ConfPath() string {
+	if firelens.firelensConfigType == FirelensConfigTypeFluentd {
+		return S3ConfigPathFluentd + "external.conf"
```

Comment:
Unfortunately not without other changes since we actually use these const values as the bind mount. Since Windows bind mounts the directory that is why I opted to handle it this way. https://github.com/vineeth-bandi/amazon-ecs-agent/blob/ab461f8845a23863ba63ab84ec175a561bc43cfa/agent/api/task/task.go#L441